DependencyInjection/Compiler/EntityListenerPass.php

```diff
@@ -105,6 +105,8 @@ private function attachToListener(ContainerBuilder $container, $name, $id, array
 
         if (isset($attributes['method'])) {
             $args[] = $attributes['method'];
+        } elseif (! method_exists($serviceDef->getClass(), $attributes['event']) && method_exists($serviceDef->getClass(), '__invoke')) {
```

Comment:
this doesn't account for ChildDefinition, I'm sure someone will complain soon :)

Reply:
Should be fixed in https://github.com/doctrine/DoctrineBundle/pull/999